src/services/codefixes/disableJsDiagnostics.ts

```diff
@@ -0,0 +1,74 @@
+/* @internal */
+namespace ts.codefix {
+    registerCodeFix({
+        errorCodes: getApplicableDiagnosticCodes(),
+        getCodeActions: getDisableJsDiagnosticsCodeActions
+    });
+
+    function getApplicableDiagnosticCodes(): number[] {
+        const allDiagnostcs = <MapLike<DiagnosticMessage>>Diagnostics;
+        return Object.keys(allDiagnostcs)
+            .filter(d => allDiagnostcs[d] && allDiagnostcs[d].category === DiagnosticCategory.Error)
+            .map(d => allDiagnostcs[d].code);
+    }
+
+    function shouldCheckJsFile(sourceFile: SourceFile, compilerOptions: CompilerOptions) {

```

Comment:
this is the same snipped with [this one](https://github.com/Microsoft/TypeScript/pull/14496/files#diff-08a3cc4f1f9a51dbb468c2810f5229d3R922). Can we share this?